packages/std/src/testing/mock.rs

```diff
@@ -1247,31 +1215,36 @@ mod tests {
     }
 
     #[test]
-    fn addr_canonicalize_min_input_length() {
+    fn addr_canonicalize_short_input() {
         let api = MockApi::default();
-        let human = String::from("1");
-        let err = api.addr_canonicalize(&human).unwrap_err();
-        assert!(err
+
+        // empty address should fail
+        let empty = "cosmwasm1pj90vm";
+        assert!(api
+            .addr_canonicalize(empty)
+            .unwrap_err()
             .to_string()
-            .contains("human address too short for this mock implementation (must be >= 3)"));
+            .contains("Invalid canonical address length"));
+
+        // one byte address should work
+        let human = "cosmwasm1qqvk2mde";
+        assert_eq!(api.addr_canonicalize(human).unwrap().as_ref(), [0u8]);
     }
 
     #[test]
-    fn addr_canonicalize_max_input_length() {
+    fn addr_canonicalize_long_input() {
         let api = MockApi::default();
         let human =
-            String::from("some-extremely-long-address-not-supported-by-this-api-longer-than-supported------------------------");
-        let err = api.addr_canonicalize(&human).unwrap_err();
-        assert!(err
-            .to_string()
-            .contains("human address too long for this mock implementation (must be <= 90)"));
+            "some-extremely-long-address-some-extremely-long-address-some-extremely-long-address-some-extremely-long-address";
+        let err = api.addr_canonicalize(human).unwrap_err();
+        assert!(err.to_string().contains("Invalid input"));
     }
 
     #[test]
-    #[should_panic(expected = "length not correct")]
+    #[should_panic(expected = "Invalid canonical address length")]
```

Comment:
It's a bit pointless to check the error through panic behaviour via `.unwrap()`. Could you check the error directly?